core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js

```diff
@@ -18,9 +18,10 @@
  */
 
 oppia.factory('AnswerGroupObjectFactory', [function() {
-  var AnswerGroup = function(ruleSpecs, outcome, correct) {
-    this.ruleSpecs = ruleSpecs;
-    this.outcome = outcome;
+  var AnswerGroup = function(ruleSpecsBackendList, outcomeBackendDict,
+    correct) {
+    this.ruleSpecs = ruleSpecsBackendList;
```

Comment:
Hm, yeah, I think maybe this should just be ruleSpecsList, outcomeDict, etc. and the createFromBackendDict() takes in a backend dict and transforms it into the right format before passing to the constructor.